qcodes/instrument/base.py

```diff
@@ -106,11 +106,21 @@ def add_parameter(self, name: str,
             **kwargs: Constructor arguments for ``parameter_class``.
 
         Raises:
-            KeyError: If this instrument already has a parameter with this
-                name.
-        """
-        if name in self.parameters:
-            raise KeyError(f'Duplicate parameter name {name}')
+            ValueError: If this instrument already has a parameter with this
+                name but its unit are not equal to the unit of the parameter
+                being added.
+        """
+        existing_parameter = self.parameters.get(name, None)
+        if existing_parameter:
+            existing_unit = getattr(existing_parameter, "unit", None)
+            new_unit = kwargs.get("unit", None)
+            if existing_unit != new_unit:
```

Comment:
I guess we should also raise here if the existing_parameter is not abstract ? Or do we want to allow non abstract parameters to be overwritten?

Reply:
I would vote for raising an exception if the parameter we wish to override is not abstract.